halo2/src/prover.rs

```diff
@@ -179,6 +207,83 @@ impl Halo2Prover {
             panic!("powdr modulus doesn't match halo2 modulus. Make sure you are using Bn254");
         }
     }
+
+    fn verification_key_inner<F: FieldElement>(
+        &self,
+        pil: &Analyzed<F>,
+        fixed: &[(String, Vec<F>)],
+    ) -> VerifyingKey<G1Affine> {
+        let (plaf_circuit, _) = analyzed_to_plaf(pil, fixed);
+        let circuit = analyzed_to_circuit_with_zeroed_witness(plaf_circuit, pil);
+        keygen_vk(&self.params, &circuit).unwrap()
+    }
+
+    pub fn verification_key<F: FieldElement>(
+        &self,
+        pil: &Analyzed<F>,
+        fixed: &[(String, Vec<F>)],
+    ) -> Vec<u8> {
+        let vk = self.verification_key_inner(pil, fixed);
+        vk.to_bytes(SerdeFormat::Processed)
+    }
+
+    pub fn verify_inner<
```

Comment:
I think I would just inline these `_inner` functions, but might be a taste thing. At least it shouldn't be public, right?

Reply:
You mean Rust `inline` or replace the code? The reason why it's a function is that it's used both at proof generation and verification times. But yes, shouldn't be public